app/src/main/java/dev/spikeysanju/expensetracker/services/csv/ExportService.kt

```diff
@@ -0,0 +1,63 @@
+package dev.spikeysanju.expensetracker.services.csv
+
+import android.os.Environment
+import androidx.annotation.WorkerThread
+import com.opencsv.CSVWriter
+import com.opencsv.bean.StatefulBeanToCsvBuilder
+import kotlinx.coroutines.flow.flow
+import java.io.File
+import java.io.FileWriter
+import java.text.DateFormat
+
+// todo
+// add permission check
+// add scoped storage
+// add test cases for file creation
+
+/****
+ * Author : Ch8n
+ * Created-on : 10-02-2021
+ */
+sealed class Exports {
+    data class CSV(val csvConfig: CsvConfig) : Exports()
+}
+
+data class CsvConfig(
+    val prefix: String = "expenso",
+    val suffix: String = DateFormat
+        .getDateTimeInstance()
+        .format(System.currentTimeMillis())
+        .toString()
+        .replace(" ", "_"),
+    val fileName: String = "$prefix-$suffix.csv",
+    val hostPath: String = Environment
+        .getExternalStorageDirectory()?.absolutePath?.plus("/Documents/Expenso") ?: ""
+)
```

Comment:
`.getExternalStorageDirectory()` - It's deprecated. Shall we improve on this? @ch8n

Reply:
Hi it's deprecated for API level 30 which needs scoped storage I have open an issue for that and currently, the app is blocked to work on this feature on API level 29+